include/sound/sof/xtensa.h

```diff
@@ -15,6 +15,9 @@
  * Architecture specific debug
  */
 
+/* AR0 ~ AR15 */
+#define XTENSA_CORE_MAX_AR_REGS_COUNT	16
+
```

Comment:
Uh, sorry @RanderWang , your original code was correct. 

I now realize xoops->plat_hdr is not directly read from the memory window, but rather parsed in kernel. On line 84 of intel/telemetry.c (you add in this patch), you already limit the AR count to 16. So this number can be trusted after all, it cannot ever be larger than 16 no matter what is in the telemetry contents.

So you can revert to the original version after all (which already had the reviews and was ready to go, apologies!).

Reply:
NP ,  reverted